src/main.js

```diff
@@ -0,0 +1,315 @@
+  let projects = JSON.parse(localStorage.getItem('projects'));
+  let lastProject = JSON.parse(localStorage.getItem('lastProject'));
+
+  if (projects == null || projects.length === 0) {
+    projects = [new Project('Your first project', [])];
+  } else {
+    const dummy = [...projects];
+    projects = [];
+    for (let i = 0; i < dummy.length; i += 1) {
+      const timmy = [];
+      dummy[i].toDo.forEach((todo) => {
+        timmy.push(
+          new ToDo(
+            todo.title,
+            todo.description,
+            todo.dueDate,
+            todo.priority,
+            todo.check,
+          ),
+        );
+      });
+      projects.push(new Project(dummy[i].title, timmy));
+    }
+  }
+
+  if (
+    projects.filter((project) => project.title === lastProject).length === 0
+  ) {
+    lastProject = projects[0].title;
+  }
+
+  // save last project
+
+  const saveLastProject = (project) => {
+    localStorage.setItem('lastProject', JSON.stringify(project.title));
+  };
+
+  // validates project is filled out
+
+  const validateProject = () => {
+    const form = document.getElementById('project-form');
+    if (form[0].value === '') {
+      return alert('Title must be filled out'); // eslint-disable-line no-alert
+    }
+    const project = new Project(form[0].value);
+
+    projects.push(project);
+    return project.save(projects);
+  };
+
+  // validates ToDo and creates a new one
+
+  const validateToDo = (project = null, toDo = null) => {
+    const form = document.getElementById('todo-form');
+    if (
+      form[0].value === ''
+      || form[1].value === ''
+      || form[2].value === ''
+      || form[3].value === ''
+    ) {
+      return alert('Form should be completely filled out.'); // eslint-disable-line no-alert
+    }
+
+    let newToDo = null;
+
+    if (toDo === null) {
+      newToDo = new ToDo(
+        form[0].value,
+        form[1].value,
+        form[2].value,
+        form[3].value,
+      );
+    } else {
+      toDo.title = form[0].value;
+      toDo.description = form[1].value;
+      toDo.dueDate = form[2].value;
+      toDo.priority = form[3].value;
+    }
+
+    if (newToDo) {
+      project.toDo.push(newToDo);
+    }
+
+    project.save(projects);
+    return saveLastProject(project);
+  };
+
+  // renders todo form
+
+  const renderForm = (project = null, toDo = null) => {
+    const validateForm = document.getElementById('todo-form');
+    if (validateForm) {
+      validateForm.parentNode.removeChild(validateForm);
+      return false;
+    }
+
+    Form.render();
+
+    const form = document.getElementById('todo-form');
+
+    const submit = document.createElement('button');
+    submit.setAttribute('type', 'submit');
+    submit.classList.add('form-button');
+    submit.onclick = () => validateToDo(project, toDo);
+    if (toDo === null) {
+      submit.innerHTML = 'Create Task';
+    } else {
+      submit.innerHTML = 'Save Changes';
+    }
+
+    return form.appendChild(submit);
+  };
```

Comment:
Let's separate the DOM from application logic, you can create another file for DOM.

![image](https://user-images.githubusercontent.com/31897434/73420341-70e18b80-4322-11ea-840a-fa14e04b592b.png)

https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59622602/

Reply:
We talked to @bellom on Slack and we agreed that this function is just creating a button in DOM and not applying any application logic.